src/source/CppGenerator.scala

```diff
@@ -91,11 +91,34 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
   }
 
   def generateHppConstants(w: IndentWriter, consts: Seq[Const]) = {
+    var needsCppGen = false
+
     for (c <- consts) {
+      var isConstExpr = true
+      val constValue = c.value match {
+        case l: Long => " = { " + l.toString + " };"
+        case d: Double if marshal.fieldType(c.ty) == "float" => " = { " + d.toString + "f };"
+        case d: Double => " = { " + d.toString + " };"
+        case b: Boolean => if (b) " = { true };" else " = { false };"
+        case e: EnumValue => " = " + marshal.typename(c.ty) + "::" + idCpp.enum(e.ty.name + "_" + e.name) + ";"
+        case _ => {
+          needsCppGen = true;
+          isConstExpr = false;
+          ";"
```

Comment:
I'd suggest making constValue empty here, and applying the semi-colon in the final statement below.  The semicolon isn't really part of the const value.